### ISSUE TITLE
Question about persistent VFX

### ISSUE THREAD

**Issue by NaesKun** [0ceal0t/Dalamud-VFXEditor#13]
Hi, I was wondering if this VFX tool has the capability to add persistent "actor" weapon VFX independently of a buff or gauge window. Specifically I'm looking to have the Gauss Barrel from the old Gauss Barrel skill stay on during the duration of a Hypercharge window which should be about 8 seconds.

I wanted to know if this was possible to do it within the tool without additional imports from outside sources and if so how would I go about doing that.

**Comment by 0ceal0t** [comment]
you should just be able to merge the gauss barrel vfx into the hypercharge one and set the lifetime to 8 seconds (or how ever many frames).

**Comment by NaesKun** [comment]
0ceal0t, this is my first time modding so bear with me. I kind of need an ELI5 explanation. Even after reading your guide, I'm still lost.

Where do I find the lifetime option, under Timelines? Also under Timelines > Parameters tab can you help me figure out what each binder is (Binder 1 Linear and Binder 0 Point). Also what does Loop Start and Loop End indicate?

What does the Emitter and the numbers indicate under Timelines > Items Tab?

**Comment by NaesKun** [comment]
I was thinking another solution to what I wanted to do:

I make the base Gauss Barrel animation persistent where the Gauss Barrel stays on the weapon for that X amount of time.
Then I adjust the Gauss Barrel Animation then do the Skill Effect Editor to swap the animation between two.

But my issue is still keeping the gauss barrel on my weapon for 8 seconds. I don't know how to keep that effect persistent.

**Comment by NaesKun** [comment]
Let me get the high level view so you understand what I'm trying to do. Sorry, I'm still working on communication.

I basically want the old Gauss Barrel VFX to replace the current Hypercharge, I want the Gauss Barrel to stay on the end of the gun for the full duration of Overheat. How would this tool help me accomplish this?

**Comment by 0ceal0t** [comment]
I don't think that the gauss barrel is a vfx. You'd probably better off using something like Skillswap (https://github.com/0ceal0t/SkillSwap) to swap their animations. It's not perfect and doesn't work with all animations, but it's worth a shot

**Comment by NaesKun** [comment]
Thanks, I'll move the thread over to that github and close this with comment.

**Comment by NaesKun** [comment]
Referencing: https://github.com/0ceal0t/SkillSwap/issues/3

So now, I got the barrel to persist and vanish matching the Hypercharge buff window. Now I'm stuck with a floating Gauss Barrel in the air for the duration. How do I "mount" or "attach" it to my weapon?

**Comment by 0ceal0t** [comment]
ok, I was wrong again 😭 the vfx isn't actually the gauss barrel attachment, it's just the short animation when it's being applied. The actual attachment is a part of the weapon model itself. I'm not sure how the game is deciding when to hide/show it, so that would take some reverse engineering, which I don't currently have a huge amount of time for.

![image](https://user-images.githubusercontent.com/18051158/140664552-e55c8958-bcc9-4156-bbd6-23441823b48c.png)

**Comment by 0ceal0t** [comment]
it might be manually checking if the "Gauss barrel" buff is active (which isn't currently possible in the game) or maybe something in the `.tmb` file is showing/hiding that part of the model. Either way, it's not something that either VFXEditor or Skillswap can really change.

**Comment by 0ceal0t** [comment]
Oh a related note, the attachment isn't part of the models of newer machinist weapons (ShB and onwards)

![image](https://user-images.githubusercontent.com/18051158/140664682-e1795898-571a-44d0-99f2-4268dc8d9d2d.png)

**Comment by 0ceal0t** [comment]
so it's likely impossible to display on them without some significant work

**Comment by NaesKun** [comment]
Thanks for the very thorough explanation. This is super enlightening.

So I'm thinking about applying it to a single weapon (Anemos Outsider) for personal use. 

Is there a way to dig into the .tmb file to see the logic or code behind what triggers it and if so import that into the VFX Editor to play with? Or would I have to actually do an edit within the .tmb file for the weapon and set a trigger condition for it to appear when Hypercharge is online and disappear when Hypercharge is not online?

**Comment by 0ceal0t** [comment]
VFXEditor should show the `.tmb` path for Gauss barrel when you search for it. You can then find that path in XIVExplorer (https://github.com/goaaats/ffxiv-explorer-fork) or extract the file manually using Textools `Tools > Raw File Operations > Extract Raw File`) and open it with a hex editor.

After that, you're a bit on your own. Like I said before, TMB files aren't very well understood. There's some documentation around, and you can probably find people more knowledgeable than me in the XIVLauncher discord.

If, however, it isn't being handled by the `.tmb` and is some custom code, you'll have to dig into the game executable using a disassembler.

**Comment by NaesKun** [comment]
Thanks for the thorough answer 0cel0t. Appreciate all the help!

**Comment by 0ceal0t** [comment]
np. Also as a final note, I am working on reverse-engineering tmb files, but it probably won't be done for a while. If I find anything helpful, I'll let you know.
